pkg/fish-spec/framework/color.fish

```diff
@@ -0,0 +1,35 @@
+function echo::success
+  set_color -o green
+  echo $argv
+  set_color normal
+end
+
+function echo::failure
+  set_color -o red
+  echo $argv
+  set_color normal
+end
+
+function echo::mixed
+  set_color -o yellow
+  echo $argv
+  set_color normal
+end
+
+function echo::info
+  set_color -o cyan
+  echo $argv
+  set_color normal
+end
+
+function echo::autocolor -a total failed
+  if test $total -eq $failed -a $total -eq 0
+    echo::success $argv[3..]
+  else if test $total -eq $failed
+    echo::failure $argv[3..]
+  else if test $failed -gt 0
+    echo::mixed $argv[3..]
+  else
+    echo::success $argv[3..]
+  end
+end
```

Comment:
we should prolly namespace these functions and prepend `__` to avoid clashes with user defined code. e.g.: 

```fish
__fish_spec_echo_autocolor
```